lib/ansible/modules/database/postgresql/postgresql_slot.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: postgresql_slot
+short_description: Add or remove slots from a PostgreSQL database.
+description:
+  - Adds or removes physical or logical slots from a PostgreSQL database.
+version_added: "2.8"
+options:
+  slot_name:
+    description:
+      - name of the slot to add or remove
+    required: true
+  type:
+    description:
+      - slots come in two distinct flavors, physical and logical
+    required: false
+    default: physical
+    choices: [ "physical", "logical" ]
+  db:
+    description:
+      - Name of the database where you're connecting in order to add or remove only the logical slot to/from
+    required: false
+  decoder:
+    description:
+      - All logical slots must indicate which decoder they're using. This parameter does not apply to physical slots.
+    required: false
+    default: "test_decoding"
+  login_user:
+    description:
+      - The username used to authenticate with
+    default: postgres
+  login_password:
+    description:
+      - The password used to authenticate with
+  login_host:
+    description:
+      - Host running the database
+    default: localhost
+  port:
+    description:
+      - Database port to connect to.
+    default: 5432
+  state:
+    description:
+      - The slot state. Whether you wish the slot to be present in the system or to not be present there.
+    default: present
+    choices: [ "present", "absent" ]
+notes:
+  - The default authentication assumes that you are either logging in as or sudo'ing to the C(postgres) account on the host.
+  - Physical replication slots were introduced to PostgreSQL with version 9.4, while logical replication slots were added beginning with version 10.0.
+    Physical slots are explained in https://www.postgresql.org/docs/9.4/protocol-replication.html and logical slots which were a later edition are
```

Comment:
Hardcoded major versions should be replaced by "current" to always provide up to date pages for end users